app/email-incoming/index.ts

```diff
@@ -0,0 +1,180 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+
+import {
+  ListUsersInGroupCommand,
+  CognitoIdentityProviderClient,
+} from '@aws-sdk/client-cognito-identity-provider'
+import { S3Client, GetObjectCommand } from '@aws-sdk/client-s3'
+import { DynamoDBAutoIncrement } from '@nasa-gcn/dynamodb-autoincrement'
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb'
+import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
+import { SESClient, SendEmailCommand } from '@aws-sdk/client-ses'
+import type { S3Event } from 'aws-lambda'
+
+import type { Source } from 'mailparser'
+import { simpleParser } from 'mailparser'
+
+import {
+  emailIsAutoReply,
+  subjectIsValid,
+  formatAuthor,
+} from '../routes/circulars/circulars.lib'
+
+import { tables } from '@architect/functions'
+import { extractAttributeRequired, extractAttribute } from '~/lib/cognito'
+
+const s3 = new S3Client({})
+const client = new CognitoIdentityProviderClient({})
+const sesClient = new SESClient({})
+const doc = DynamoDBDocument.from(new DynamoDBClient({}))
+
+const USER_POOL_ID = process.env.USER_POOL_ID
+
+const domain =
+  process.env.NODE_ENV === 'development'
+    ? 'dev.'
+    : process.env.NODE_ENV === 'test'
+    ? 'test.'
```

Comment:
`process.env.NODE_ENV` is always `production` in the deployed Lambda, regardless of the domain.

Reply:
Ah okay, is there a property that tracks the domain? I know we use `new URL(request.url)` elsewhere, but I assumed that would not work in this case.

Reply:
I don't know. Is there anything in the Lambda payload?

Reply:
I dont see any properties [on the docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html) that jump out specifically as having that detail. Is there a way to tell from the bucket? Like from the name or arn? or are those the same across environments?

Reply:
No, I see there is nothing of use in the Lambda payload.

Reply:
What if you add an environment variable to provide the email domain?